src/backend/pagemap.h

```diff
@@ -19,84 +20,143 @@ namespace snmalloc
   private:
     static constexpr size_t SHIFT = GRANULARITY_BITS;
 
-    // Before init is called will contain a single entry
-    // that is the default value.  This is needed so that
-    // various calls do not have to check for nullptr.
-    //   free(nullptr)
-    // and
-    //   malloc_usable_size(nullptr)
-    // do not require an allocation to have ocurred before
-    // they are called.
+    /**
+     * Before init is called will contain a single entry
+     * that is the default value.  This is needed so that
+     * various calls do not have to check for nullptr.
+     *   free(nullptr)
+     * and
+     *   malloc_usable_size(nullptr)
+     * do not require an allocation to have ocurred before
+     * they are called.
+     */
     inline static const T default_value{};
+
+    /**
+     * The representation of the page map.
+     *
+     * Initially a single element to ensure nullptr operations
+     * work.
+     */
     T* body{const_cast<T*>(&default_value)};
 
+    /**
+     * If `has_bounds` is set, then these should contain the
+     * bounds of the heap that is being managed by this pagemap.
+     */
     address_t base{0};
     size_t size{0};
 
+  public:
     /**
-     * Commit entry
+     * Ensure this range of pagemap is accessible
      */
-    void commit_entry(void* p)
+    void register_range(address_t p, size_t length)
```

Comment:
Should this be gated, either with `enable_if` or at all call sites, on `!pal_supports<LazyCommit>`?  Somewhat relatedly, do we ever `notify_not_using` any part of the pagemap?

Reply:
So I am not sure about that.  On POSIX, with `SNMALLOC_CHECK_CLIENT`, we are using `PROT_NONE` a lot, so the `using` and `not_using` is required.  I could change `LazyCommit` to be off for POSIX with `SNMALLOC_CHECK_CLIENT`.  Then I could potentially gate these things.  At the moment, I am just hoping the empty bodied functions get inlined, and erased to not cost when we are exploiting `LazyCommit.`